lib/providers.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'todo_item.dart';
+
+final todoAppProvider =
+    ChangeNotifierProvider<AppChangeNotifier>((ref) => AppChangeNotifier());
+
+class AppChangeNotifier extends ChangeNotifier {
+  final List<TodoItem> unCompletedList = [
+    TodoItem(
+      id: 0,
+      title: 'dart',
+      content: 'dart-content',
+      isCompleted: false,
+    ),
+    TodoItem(
+      id: 1,
+      title: 'ssh',
+      content: 'ssh',
+      isCompleted: false,
+    ),
+  ];
+  final List<TodoItem> completedList = [
+    TodoItem(
+      id: 2,
+      title: 'python',
+      content: '再帰関数',
+      isCompleted: true,
+    ),
+    TodoItem(
+      id: 3,
+      title: 'flutter',
+      content: '非同期処理',
+      isCompleted: true,
+    ),
+  ];
+
+  int bottomIndex = 0;
+
+  void bottomIndexChange(int index) {
+    bottomIndex = index;
+    notifyListeners();
+  }
+
+  void addTodoItem(Map<String, String> formValue) {
+    unCompletedList.add(
+      TodoItem(
+        id: unCompletedList.length + completedList.length,
+        title: formValue['title']!,
+        content: formValue['content']!,
+        isCompleted: false,
+      ),
+    );
+    notifyListeners();
+  }
```

Comment:
この `AppNotifier` は `TodoItem` を管理しているものであって、フォームとは無関係なはずなので、`TodoItem` をそのまま受け取る方がいいと多います。作成ページ側で `TodoItem` のインスタンスを作って渡す方が安全ですね。
```suggestion
  void addTodoItem(TodoItem todo) {
    unCompletedList.add(todo);
    notifyListeners();
  }
```